httptrace/client.go

```diff
@@ -174,3 +175,43 @@ func cloneRequest(r *http.Request) *http.Request {
 	}
 	return r2
 }
+
+// saveCloneRequest keeps clone request for CancelRequest
+func (t *Transport) saveCloneRequest(original *http.Request, clone *http.Request) {
+	t.reqMu.Lock()
+	defer t.reqMu.Unlock()
+	if t.requests == nil {
+		t.requests = make(map[*http.Request]*http.Request)
+	}
+
+	if clone != nil {
+		t.requests[original] = clone
+	} else {
+		delete(t.requests, original)
+	}
+}
+
+// getTransport returns custom transport or DefaultTransport
+func (t *Transport) getTransport() http.RoundTripper {
+	if t.Transport != nil {
+		return t.Transport
+	}
+	return http.DefaultTransport
+}
+
+// CancelRequest cancels an in-flight request by closing its connection.
+func (t *Transport) CancelRequest(req *http.Request) {
+	type canceler interface {
+		CancelRequest(*http.Request)
+	}
+
+	transport := t.getTransport()
```

Comment:
`transport` is not used as part of the if statement below, so move it further down around line 214